### ISSUE TITLE
Bug fix in EspMilightHubHandler.java:137 ??

### ISSUE THREAD

**Issue by koimotion** [Skinah/EspMilightHub#13]
Hi,

Great to have the ESP hub and your binding available. I have a lot of cct and rgb+cct lights all working fine via remotes after a major renovation. Getting Openhab 2.5 running now after tests were done a few years back with the Milight binding and an ibox. The ESP hub and MQTT are working fine apart from some RF antenna issues I should improve soon. I can control the lights from the ESP hub web interface, PaperUI, BasicUI and Android app. However when I use a sitemap slider to change the cct colour temp (warm white vs cool white) I get:

16:58:14.859 [INFO ] [smarthome.event.ItemCommandEvent     ] - Item 'CorridorPendantsColourtemperature' received command 10.0
16:58:14.899 [INFO ] [arthome.event.ItemStatePredictedEvent] - CorridorPendantsColourtemperature predicted to become 10.0
16:58:14.908 [ERROR] [rnal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.espmilighthub.handler.EspMilightHubHandler@249cf9': For input string: "10.0"
java.lang.NumberFormatException: For input string: "10.0"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65) ~[?:1.8.0_222]
	at java.lang.Integer.parseInt(Integer.java:580) ~[?:1.8.0_222]
	at java.lang.Integer.valueOf(Integer.java:766) ~[?:1.8.0_222]
	at org.openhab.binding.espmilighthub.handler.EspMilightHubHandler.handleCommand(EspMilightHubHandler.java:137) ~[?:?]

I made up a test java case with Float instead of Integer.valueOf(command.toString

public class Test {
    public static void main(String[] args) {
        String x = "10.0";
        System.out.println(x);
        int scaledCommand = (int) Math.round((370 - (2.17 * Float.valueOf(x.toString()))));
        System.out.println(scaledCommand);
    }
}

This gives no error while your code of Integer.valueOf from github gives the same error using a sitemap slider to change colour temp. I tried to build a new jar file with this change and ran into lots of maven errors and am unsure as to how you make you jar and what is needed for the new maven + bnd environment.

Can you comment and advise how I could build a new jar

Thanks for the great work, Andrew

**Comment by Skinah** [comment]
I use this command to build
mvn clean compile package

or you can do mvn clean install with the dskcheckskip (?) option

**Comment by koimotion** [comment]
I was using VSCode and Maven commands and then did a new install of Eclipse and followed the Openhab IDE guidelines and have openhab-2-5-x git directories for openhab-addons, openhab-distro and EspMilightHub (from a fork and git clone of your work). Initially I had tons of errors and then changed your pom to 2.5.3-SNAPSHOT to get rid of the errors
  <parent>
    <groupId>org.openhab.addons.bundles</groupId>
    <artifactId>org.openhab.addons.reactor.bundles</artifactId>
    <version>2.5.3-SNAPSHOT</version>
  </parent>
The jar starts to be built and then fails using many attempts and suggestions from the openhab community. For example
mvn clean install -U -am -DskipChecks -DskipTests -X
All attempts give
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.722 s
[INFO] Finished at: 2020-03-12T15:23:13+11:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.commonjava.maven.plugins:directory-maven-plugin:0.3.1:directory-of (directories) on project org.openhab.binding.espmilighthub: Cannot find directory for project: org.openhab.addons:org.openhab.addons.reactor 
What do the parent pom commands point to on your setup?
Should I use an alternate parent?
Thanks

**Comment by koimotion** [comment]
Hi, I have finally had success. I used the skeleton.sh to build a newtest binding (it was put under git/openhab-addons/bundles/org.openhab.binding.newtest) and built a jar just fine using mvn clean install. It has basically the same he same pom as yours. I then made a directory git/openhab-addons/bundles/org.openhab.binding.espmilighthub and put your files in this directory

~/openhab-2-5-x/git/openhab-addons/bundles/org.openhab.binding.espmilighthub$ ls -a
.  ..  .classpath  .git  .gitignore  LICENSE  NOTICE  pom.xml  .project  README.md  .settings  src  target
mvn clean install

I did get BUILD FAILURE, due to correctness and style errors, but the the jar works!

Even better, using a slider in the openhab app to control the cct pendants from cool white to warm white and back NOW WORKS when I put the 2.5.3 jar on my openhabian rp3.

Note that when I used git clone it made a directory EspMilightHub and the build failed as previously. By getting rid of EspMilightHub it worked.

I will do further tests on both cct and RGB+cct lights before putting the code fix request in via git

Andrew
